### DISCUSSION TITLE
When are Maven Dependencies resolved to the .m2 cache versus the workspace project

### DISCUSSION THREAD

**Discussion by merks** [eclipse-m2e/m2e-core#1777] [Q&A] [answered]
I noticed the following issue in Mylyn and those folks had to jump through hoops to address this problem, i.e, they must do a full local mvn build to avoid a workspace with errors.

Specifically we see here the pom org.eclipse.mylyn.wikitext.maven with two dependency on other "projects" in the same git repository:

![image](https://github.com/eclipse-m2e/m2e-core/assets/208716/3ad41fb9-caf0-48b2-9744-c2dc71579096)

In the Maven Dependencies of this project these two dependencies are bound to jars in the .m2 cache.   But if I navigate the dependency (Ctrl-Click), it navigates to the pom of the workspace project.   So I would expect the Maven Dependencies to be bound to the workspace projects for these two dependencies.   Is that a misguided expectation?  In other projects on which I worked, I do see that it's possible to map to a project:

![image](https://github.com/eclipse-m2e/m2e-core/assets/208716/4c65795b-b381-4d04-a3b9-002ca5b0ef38)

So I don't understand how this works.  It seems really important to bind to the workspace project for a dependency when that is available in the workspace. 

Is this a bug?

**Comment by davidmichaelkarr** [comment]
You appear to be referring to "Workspace Resolution", which can be turned on or off in the "Maven" submenu  of the context menu for the project. In fact, I often do NOT want workspace resolution enabled. I'm often working on experimental versions of libraries, and I don't want applications to automatically use that. I want them to use what's in the local maven repo. Unfortunately, I find that sometimes the "Workspace Resolution" flag is ignored, and applications appear to bind to workspace projects even when I have that flag turned off.

**Reply by merks** [reply]
Apparently it is enabled:

![image](https://github.com/eclipse-m2e/m2e-core/assets/208716/d9dd8afb-4379-407c-864d-bd6a28793836)

And when I disabled it, there were errors.

So I  (Mylyn) appear to have the opposite problem, i.e., no matter what I do, it doesn't resolve to the workspace projects which I definitely want...

**Reply by laeubi** [reply, marked as answer]
At laest in your screenshot above the `mylin.wikitext` project is not marked as being a maven project so it won't be discovered but its a bit hard to follow what should be where... but in general if you import it "only" as a java project m2e will not consider it for workspace resolution.

**Reply by merks** [reply]
Yes, that explains it, so I can get it to look like this:

![image](https://github.com/eclipse-m2e/m2e-core/assets/208716/22081658-3771-4a0f-9008-6531124024e5)
